.github/workflows/macos.yml

```diff
@@ -71,3 +71,32 @@ jobs:
         uses: actions-rs/cargo@v1
         with:
           command: test
+
+  uniffi-check:
+    # macos-*-xlarge is needed to get M1 chips - https://github.blog/2023-10-02-introducing-the-new-apple-silicon-powered-m1-macos-larger-runner-for-github-actions/
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
```

Comment:
This can use `${{ env.TOOLCHAIN }}` and in the original.  I haven't made that change yet, but it does help with duplication.